artman/conductors/cloudtasks_conductor.py

```diff
@@ -0,0 +1,147 @@
+        name=queue_name, body=body).execute()
+    logger.info('Pulling tasks request returned %s' % tasks)
+    return tasks
+
+
+def _ack_task(task_client, task):
+    body = {'scheduleTime': task['scheduleTime']}
+    response = task_client.projects().locations().queues().tasks().acknowledge(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Acknowledge task request returned %s' % response)
+    return response
+
+
+def _cancel_task_lease(task_client, task):
+    body = {'scheduleTime': task['scheduleTime'], "responseView": "FULL"}
+    response = task_client.projects().locations().queues().tasks().cancelLease(
+        name=task['name'],
+        body=body).execute()
+    logger.info('Cancel task request returned %s' % response)
+    return response
+
+
+def _execute_task(artman_user_config, task):
+    """Execute the remote artman tasks.
+
+    It execute the artman command with a customized artman user config and
+    additional pipeline arguments."""
+    task_payload = base64.b64decode(task['pullTaskTarget']['payload'])
+    artman_args = task_payload.decode("utf-8").split(' ')
+    artman_args.append('--user-config')
+    artman_args.append(artman_user_config)
+    # TODO(ethanbao): No longer need to override toolkit_path once the
+    # toolkit_path config in artman user config won't be overridden with
+    # ${REPOROOT}/toolkit
+    if os.environ.get('TOOLKIT_HOME'):
+        artman_args.append('--pipeline-kwargs')
+        artman_args.append(
+            '{"toolkit_path": "%s"}' % os.environ.get('TOOLKIT_HOME'))
+    main.main(*artman_args)
+
+
+def _prepare_dir(source_repo="https://github.com/googleapis/googleapis.git"):
+    """Prepare the temporary folder to task execution.
+
+    It downloads the googleapis repo and adds a one-time artman config yaml.
+    TODO(ethanbao): support loading more input files from heterogeneous data
+    sources"""
+
+    repo_root = '/tmp/artman/%s' % str(uuid.uuid4())
+    logger.info('Prepare a temporary root repo: %s' % repo_root)
+    try:
+        os.makedirs(repo_root)
+    except OSError as e:
+        raise e
+    logger.info('Checking out fresh clone of %s.' % source_repo)
+    googleapis_dir = os.path.join(repo_root, "googleapis")
+    git_clone_args = ['git', 'clone', source_repo, googleapis_dir]
+    output = subprocess.check_output(git_clone_args)
+    if output:
+        output = output.decode('utf8')
+        logger.info(output)
+
+    artman_user_config = os.path.join(repo_root, 'artman-config.yaml')
+    with io.open(artman_user_config, 'w+') as file_:
+        file_.write(u'---\n')
+        file_.write(u'local_paths:\n')
+        file_.write(u'  reporoot: %s\n' % repo_root)
+        if os.environ.get('TOOLKIT_HOME'):
+            toolkit_home = os.environ.get('TOOLKIT_HOME')
+            file_.write(u'  toolkit: %s \n' % toolkit_home)
+            file_.write(u'toolkit_path: %s \n' % toolkit_home)
```

Comment:
How does `toolkit_path` differ from `reporoot.toolkit`? It seems like they should not both be things.